servant-docs/src/Servant/Docs/Internal.hs

```diff
@@ -920,155 +962,22 @@ instance HasDocs sublayout => HasDocs (Vault :> sublayout) where
-         , ToSample f f
-         ) => ToSample (a, b, c, d, e, f) (a, b, c, d, e, f) where
-    toSample _ = (,,,,,)
-        <$> toSample (Proxy :: Proxy a)
-        <*> toSample (Proxy :: Proxy b)
-        <*> toSample (Proxy :: Proxy c)
-        <*> toSample (Proxy :: Proxy d)
-        <*> toSample (Proxy :: Proxy e)
-        <*> toSample (Proxy :: Proxy f)
-    toSamples _ = render
-        <$> toSamples (Proxy :: Proxy a)
-        <*> toSamples (Proxy :: Proxy b)
-        <*> toSamples (Proxy :: Proxy c)
-        <*> toSamples (Proxy :: Proxy d)
-        <*> toSamples (Proxy :: Proxy e)
-        <*> toSamples (Proxy :: Proxy f)
-      where render (ta, va) (tb, vb) (tc, vc) (td, vd) (te, ve) (tf, vf)
-              = ("("  <> ta <>
-                 ", " <> tb <>
-                 ", " <> tc <>
-                 ", " <> td <>
-                 ", " <> te <>
-                 ", " <> tf <>
-                 ")"
-                , (va, vb, vc, vd, ve, vf))
-
-instance ( ToSample a a
-         , ToSample b b
-         , ToSample c c
-         , ToSample d d
-         , ToSample e e
-         , ToSample f f
-         , ToSample g g
-         ) => ToSample (a, b, c, d, e, f, g) (a, b, c, d, e, f, g) where
-    toSample _ = (,,,,,,)
-        <$> toSample (Proxy :: Proxy a)
-        <*> toSample (Proxy :: Proxy b)
-        <*> toSample (Proxy :: Proxy c)
-        <*> toSample (Proxy :: Proxy d)
-        <*> toSample (Proxy :: Proxy e)
-        <*> toSample (Proxy :: Proxy f)
-        <*> toSample (Proxy :: Proxy g)
-    toSamples _ = render
-        <$> toSamples (Proxy :: Proxy a)
-        <*> toSamples (Proxy :: Proxy b)
-        <*> toSamples (Proxy :: Proxy c)
-        <*> toSamples (Proxy :: Proxy d)
-        <*> toSamples (Proxy :: Proxy e)
-        <*> toSamples (Proxy :: Proxy f)
-        <*> toSamples (Proxy :: Proxy g)
-      where render (ta, va) (tb, vb) (tc, vc) (td, vd) (te, ve) (tf, vf) (tg, vg)
-              = ("("  <> ta <>
-                 ", " <> tb <>
-                 ", " <> tc <>
-                 ", " <> td <>
-                 ", " <> te <>
-                 ", " <> tf <>
-                 ", " <> tg <>
-                 ")"
-                , (va, vb, vc, vd, ve, vf, vg))
+instance (ToSample a a, ToSample b b) => ToSample (a, b) (a, b)
+instance (ToSample a a, ToSample b b, ToSample c c) => ToSample (a, b, c) (a, b, c)
+instance (ToSample a a, ToSample b b, ToSample c c, ToSample d d) => ToSample (a, b, c, d) (a, b, c, d)
+instance (ToSample a a, ToSample b b, ToSample c c, ToSample d d, ToSample e e) => ToSample (a, b, c, d, e) (a, b, c, d, e)
+instance (ToSample a a, ToSample b b, ToSample c c, ToSample d d, ToSample e e, ToSample f f) => ToSample (a, b, c, d, e, f) (a, b, c, d, e, f)
+instance (ToSample a a, ToSample b b, ToSample c c, ToSample d d, ToSample e e, ToSample f f, ToSample g g) => ToSample (a, b, c, d, e, f, g) (a, b, c, d, e, f, g)
+
+instance ToSample a a => ToSample (Maybe a) (Maybe a)
```

Comment:
instances for `Const a b` (given `a`); maybe for all the monoid newtypes (`All`, `Any`, `Sum`, `Product`, `First`, `Last`, `Dual`)?